addons/account/models/account_report.py

```diff
@@ -143,7 +143,7 @@ def _validate_root_report_id(self):
     @api.constrains('line_ids')
     def _validate_parent_sequence(self):
         previous_lines = self.env['account.report.line']
-        for line in self.line_ids:
+        for line in self.line_ids.sorted('sequence'):
```

Comment:
I set up debugging, and I noticed that the new record (with a sequence < the sequence of the parent) is at the end
=> Passed the constraint
So, I sorted it.

Reply:
I don't get what you mean. Please be more explicit. Also, This looks like a distinct fix, which should be done in  a distinct commit.